flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/source/FileSourceFunction.java

```diff
@@ -46,39 +51,25 @@ public FileSourceFunction(InputFormat<String, ?> format, TypeInformation<String>
 	}
 
 	@Override
+	@SuppressWarnings("unchecked")
 	public void open(Configuration parameters) throws Exception {
 		StreamingRuntimeContext context = (StreamingRuntimeContext) getRuntimeContext();
 		this.provider = context.getInputSplitProvider();
 		inputFormat.configure(context.getTaskStubParameters());
+		serializer = typeInfo.createSerializer(getRuntimeContext().getExecutionConfig());
```

Comment:
But the code was like this before my change. Then we should work on getting a proper ExecutionConfig over to runtime.